credentials/src/test/kotlin/web5/sdk/credentials/PresentationExchangeTest.kt

```diff
@@ -578,4 +579,39 @@ class Web5TestVectorsPresentationExchange {
       assertEquals(vector.output!!.selectedCredentials, selectedCreds)
     }
   }
+  data class CreatePresFromCredTestInput(
+    val presentationDefinition: PresentationDefinitionV2,
+    val credentialJwts: List<String>
+  )
+
+  data class CreatePresFromCredTestOutput(
+    val presentationSubmission: PresentationSubmission
+  )
+
+  @Test
+  fun create_presentation_from_credentials() {
+    val typeRef = object : TypeReference<TestVectors<CreatePresFromCredTestInput, CreatePresFromCredTestOutput>>() {}
+    val testVectors = mapper.readValue(
+      File("../test-vectors/presentation_exchange/create_presentation_from_credentials.json"),
+      typeRef
+    )
+
+    testVectors.vectors.forEach { vector ->
+      val presSubmission = PresentationExchange.createPresentationFromCredentials(
+        vector.input.credentialJwts,
+        vector.input.presentationDefinition
+      )
+
+      val vectorOutputPresSubmission = vector.output!!.presentationSubmission
+
+      assertEquals(vectorOutputPresSubmission.definitionId, presSubmission.definitionId)
+      assertEquals(vectorOutputPresSubmission.descriptorMap.size, presSubmission.descriptorMap.size)
+
+      for (i in vectorOutputPresSubmission.descriptorMap.indices) {
+        assertEquals(vectorOutputPresSubmission.descriptorMap[i].id, presSubmission.descriptorMap[i].id)
+        assertEquals(vectorOutputPresSubmission.descriptorMap[i].format, presSubmission.descriptorMap[i].format)
+        assertEquals(vectorOutputPresSubmission.descriptorMap[i].path, presSubmission.descriptorMap[i].path)
+      }
```

Comment:
Would this work? 
```suggestion
      assertEquals(vectorOutputPresSubmission, presSubmission, vector.description)
```

Reply:
nope:
```
Expected :web5.sdk.credentials.model.PresentationSubmission@62517709
Actual   :web5.sdk.credentials.model.PresentationSubmission@1fcdf8c1
```

I bet there is a deep equals or something let me look a bit

Reply:
That likely means you could make it work by overriding the equals method.

Reply:
making it a data class would fix that otherwise have to do the override hashcode / equals stuff.

However this still wont work in this case because the presSubmission ids will be different (by design) 

creating a new presentation definition makes a different presSubmission id guid

Reply:
> making it a data class would fix that otherwise have to do the override hashcode / equals stuff.

Yes, that's correct! The problem is that [binary backwards compatibility is broken](https://kotlinlang.org/docs/jvm-api-guidelines-backward-compatibility.html#don-t-use-data-classes-in-an-api) when using dataclasses.

> However this still wont work in this case because the presSubmission ids will be different (by design)

Ahhh, excellent point. It does feel a bit strange to have a test vector that's not compared exactly, but I guess that's fine as long as it's documented in the README!